### PR TITLE
Strip leading and trailing white space from zip code

### PR DESCRIPTION
**Pull request by mvz**
#6572.

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage to preview changes. <a href='http://mvz-issue-6572-strip-zip.spree-spree-9ed724b77adec04c49c2.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://mvz-issue-6572-strip-zip.spree-spree-9ed724b77adec04c49c2.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

**Comment by mvz** [comment]
I emulated the layout of the other code in the files I changed, but it looks like @houndci disagrees that. Should I follow its advice or leave things as they are?

**Comment by JDutil** [comment]
@mvz would be best to update fixes for hound comments. Also should we change this to stripe all the submitted params rather than just zip? I think we'd want it to strip every param not just zip.

**Comment by BenMorganIO** [comment]
I would probably recommend a `before_save` in the model. This way users that want this functionality and they aren't using the Spree Frontend will get it. Also, what if an Admin on the Spree Backend adds whitespace?

Food for thought.

Thanks @mvz!!!

**Comment by BenMorganIO** [comment]
@mvz Spree, just like Rails and Ruby, is a heaping pile of legacy code :D. @houndci was added to help lower the amount of cooks in the kitchen, or rather, start making Spree look like it was made by one cook, and not hundreds. Basically, you updated an old file that hasn't been bitten by hound yet. I'd definitely appreciate it if you could update the Hound bites and maybe other areas of the file as well.

**Comment by mvz** [comment]
> I would probably recommend a before_save in the model.

I'm not sure how that interacts with the post code check, which is what started all this. Perhaps I should go a different route and change the post code check itself to ensure it works both in frontend and backend?

>  I'd definitely appreciate it if you could update the Hound bites and maybe other areas of the file as well.

Ok, I will do that.

**Comment by mvz** [comment]
I've decided to move the fix to the Address validation, so the behavior is the same no matter where an address is created or updated.

Regarding a general solution for unwanted whitespace, a gem like [auto_strip_attributes](https://github.com/holli/auto_strip_attributes) is probably the way to go.

File: frontend/spec/controllers/spree/checkout_controller_spec.rb [hunk 1]
```diff
@@ -390,16 +396,16 @@ def spree_post_address
       end
 
       it "should set flash message for no inventory" do
-        expect(flash[:error]).to eq(Spree.t(:inventory_error_flash_for_insufficient_quantity , :names => "'#{product.name}'" ))
+        expect(flash[:error]).to eq(Spree.t(:inventory_error_flash_for_insufficient_quantity, names: "'#{product.name}'"))
```
Comment (BenMorganIO) [inline frontend/spec/controllers/spree/checkout_controller_spec.rb]: Well, its shorter...

**Comment by JDutil** [comment]
Thank you

**Comment by mvz** [comment]
Thanks!

**Pull request closed without merging**